# Include Slackware specific compile flags, provide symlinks for
# backwards compatibility and build also libqhull.so
# Copyright (C) 2016 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# Visit SlackPack at http://sotirov-bg.net/slackpack/
#
# $Id: qhull-2015.2.diff,v 1.1 2016/07/15 20:36:30 gsotirov Exp $
#
diff -urNad qhull-2015.2-orig/Makefile qhull-2015.2/Makefile
--- qhull-2015.2-orig/Makefile	2016-01-19 02:24:57.000000000 +0200
+++ qhull-2015.2/Makefile	2016-07-15 23:31:37.000000000 +0300
@@ -93,11 +93,11 @@
 #for Gnu's gcc compiler, -O3 for optimization, -g for debugging, -pg for profiling
 # -fpic needed for gcc x86_64-linux-gnu.  Not needed for mingw
 CC        = gcc
-CC_OPTS1  = -O3 -ansi -Isrc -fpic $(CC_WARNINGS)
+CC_OPTS1  = -O3 -ansi -Isrc -fpic $(CC_WARNINGS) $(SLKCFLAGS)
 CXX       = g++
 
 # libqhullcpp must be before libqhull_r
-CXX_OPTS1 = -O3 -Isrc/ $(CXX_WARNINGS)
+CXX_OPTS1 = -O3 -Isrc/ $(CXX_WARNINGS) $(SLKCXXFLAGS)
 
 # for shared library link
 CC_OPTS3  =
@@ -149,7 +149,8 @@
 # Default targets for make
 
 all: bin-lib bin/rbox bin/qconvex bin/qdelaunay bin/qhalf bin/qvoronoi bin/qhull bin/testqset \
-     bin/testqset_r qtest bin/user_eg2 bin/user_eg3 bin/user_eg qconvex-prompt
+     bin/testqset_r qtest bin/user_eg2 bin/user_eg3 bin/user_eg qconvex-prompt \
+     lib/libqhull.$(SO)
 
 help:
 	head -n 75 Makefile
@@ -222,6 +223,8 @@
 	cp src/libqhull_r/*.h src/libqhull_r/*.htm $(INCDIR)/libqhull_r
 	cp src/libqhullcpp/*.h $(INCDIR)/libqhullcpp
 	cp src/qhulltest/*.h $(INCDIR)/libqhullcpp
+	cd $(INCDIR) && ln -s libqhull qhull
+	cd $(INCDIR)/libqhull && ln -s libqhull.h qhull.h
 
 new:	cleanall all
 
@@ -635,6 +638,10 @@
 	$(CC) -shared -o $@ $(CC_OPTS3) $^
 	cd lib && ln -f -s libqhull_r.$(SO) libqhull_r.so
 
+lib/libqhull.$(SO): $(LIBQHULLS_RBOX_OBJS)
+	$(CC) -shared -o $@ $(CC_OPTS3) $^
+	cd lib && ln -f -s libqhull.$(SO) libqhull.so
+
 # don't use ../qconvex.	 Does not work on Red Hat Linux
 bin/qconvex: src/qconvex/qconvex.o lib/libqhullstatic.a
 	$(CC) -o $@ $< $(CC_OPTS2) -Llib -lqhullstatic -lm
